fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -1437,12 +1530,19 @@ private LogicalPlan withProjection(LogicalPlan input, SelectColumnClauseContext
                             expressions, input, isDistinct);
                 } else {
                     List<NamedExpression> projects = getNamedExpressions(selectCtx.namedExpressionSeq());
-                    return new LogicalProject<>(projects, ImmutableList.of(), input, isDistinct);
+                    if (containsWindowExpressions(projects)) {
+                        return new LogicalWindow<>(projects, input);
+                    }
+                    return new LogicalProject<>(projects, Collections.emptyList(), input, isDistinct);
```

Comment:
maybe we should generate LogicalWindow after we do bind.
we chould just treat windowExpression as a scalar expression.
- if we use window function without agg. Then window function will in Projections.
we could generate LogicalProject(LogicalWindow(LogicalProject)) to handle all case.
- if we use window function in agg. Then window function will in Agg's output expressions. After normalize agg it will present in the Project topping on Aggregate. and then we could do the same thing as first scene.

Reply:
nice idea~